public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -19,12 +19,12 @@ import { PanelModel } from 'app/features/dashboard/state';
 import { PanelQueryRunner } from '../query/state/PanelQueryRunner';
 
 class MetricsPanelCtrl extends PanelCtrl {
-  datasource: DataSourceApi;
+  datasource?: DataSourceApi;
   contextSrv: ContextSrv;
   datasourceSrv: any;
   timeSrv: any;
   templateSrv: any;
-  range: TimeRange;
+  range?: TimeRange;
```

Comment:
Hmm, I wonder if this is a case of https://github.com/grafana/grafana/pull/33012#discussion_r613806336